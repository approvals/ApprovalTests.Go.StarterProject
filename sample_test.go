package starterpackage

import (
	"fmt"
	approvals "github.com/approvals/go-approval-tests"
	"testing"
)

func TestNormalTest(t *testing.T) {
	if 5 != 5 {
		t.Fatal("5 != 5")
	}
}

func TestBasicApproval(t *testing.T) {
	approvals.VerifyString(t, "Hello Approvals")
}

func TestJsonApproval(t *testing.T) {
	person := Person{
		"John Galt", 100,
	}
	approvals.VerifyJSONStruct(t, person)
}

func TestCombinationsApproval(t *testing.T) {
	arr := []int{10, 20, 30, 40}
	names := []string{"john", "paul", "mary"}
	approvals.VerifyAllCombinationsFor2(t, "File Names", func(a interface{}, b interface{}) string { return fmt.Sprintf("%v_%v.txt", a, b) }, names, arr)
}
