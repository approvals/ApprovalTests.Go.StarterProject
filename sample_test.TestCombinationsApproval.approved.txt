File Names


[john,10] => john_10.txt
[john,20] => john_20.txt
[john,30] => john_30.txt
[john,40] => john_40.txt
[paul,10] => paul_10.txt
[paul,20] => paul_20.txt
[paul,30] => paul_30.txt
[paul,40] => paul_40.txt
[mary,10] => mary_10.txt
[mary,20] => mary_20.txt
[mary,30] => mary_30.txt
[mary,40] => mary_40.txt